test-suite/squidconf/status-dub.conf.instructions

```diff
@@ -0,0 +1 @@
+expect-message WARNING:.*is.a.subrange.of.*Because.of.this.*isignored
```

Comment:
> a simple way to catch mismatching instruction files

If we are going to keep to the "bad-" for negative tests, we can require that each "bad-" had an instruction file - that should be easy to adjust, I think. If positive tests (i.e., all other ones) do not need instruction files, then we can also prohibit such files for them.

Reply:
The "bad" prefix is for test cases that contain problematic configurations that should not be used as a starting point for deployed configurations. I expect all of them to have instruction files because Squid should warn about problems (at least). If this is easy to check, we probably should, but it is not a high priority, and I am afraid that it may trigger wasteful discussions. Let's not do that just yet.

The lack of a "bad" prefix does not imply the lack of instructions file. Some instructions are purely "positive".

Ideally, we should catch instruction files that do not have the corresponding configuration file. _That_ check, AFAICT, will catch most of the problems that are very difficult for a human to spot during review (and that do not result in "make check" failures today). Perhaps there is some simple Makefile.am (dependency) trick that can _require_ a configuration file for each instruction file?

Reply:
> Ideally, we should catch instruction files that do not have the corresponding configuration file. That check, AFAICT, will catch most of the problems that are very difficult for a human to spot during review (and that do not result in "make check" failures today). Perhaps there is some simple Makefile.am (dependency) trick that can require a configuration file for each instruction file?

If you know of such a trick, let's discuss adding it (in a dedicated minimal PR). There is no rush with this, of course. I just did not want this accidentally discovered problem to be forgotten _if_ it has a simple solution.

Reply:
LGTM. Please post a dedicated _official_ "Maintenance: Detect mismatching .instructions files" PR with the following adjustments:

* Rename instrlist to instructionFIles and instr to instructionFile
* If possible, use backticks for basenane invocation. This is for consistency sake: `git grep basename '*.am'`. If you prefer, save the basename result into a local cfgBasename variable and use that variable to set `cfg`.
* If possible, start with the new instrlist assignment/loop, keeping the old `@failed=...rm...` line (below the new loop) intact.
* Quit on the first error to simplify _and_ avoid producing misleading results when running tests with missing/misspelled instruction files.